Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -0,0 +1,556 @@
+                store.removeValue(forKey: counter.name)
+            case .counterWithLabels(let labelNames, var dimensions):
+                let labelsKey = LabelsKey(counter.labels)
+                guard dimensions[labelsKey] === counter else { return }
+                dimensions.removeValue(forKey: labelsKey)
+                store[counter.name] = .counterWithLabels(labelNames, dimensions)
+            default:
+                return
+            }
+        }
+    }
+
+    public func destroyGauge(_ gauge: Gauge) {
+        self.box.withLockedValue { store in
+            switch store[gauge.name] {
+            case .gauge(let storedGauge):
+                guard storedGauge === gauge else { return }
+                store.removeValue(forKey: gauge.name)
+            case .gaugeWithLabels(let labelNames, var dimensions):
+                let dimensionsKey = LabelsKey(gauge.labels)
+                guard dimensions[dimensionsKey] === gauge else { return }
+                dimensions.removeValue(forKey: dimensionsKey)
+                store[gauge.name] = .gaugeWithLabels(labelNames, dimensions)
+            default:
+                return
+            }
+        }
+    }
+
+    public func destroyTimeHistogram(_ histogram: TimeHistogram) {
+        self.box.withLockedValue { store in
+            switch store[histogram.name] {
+            case .timeHistogram(let storedHistogram):
+                guard storedHistogram === histogram else { return }
+                store.removeValue(forKey: histogram.name)
+            case .timeHistogramWithLabels(let labelNames, var dimensions, let buckets):
+                let dimensionsKey = LabelsKey(histogram.labels)
+                guard dimensions[dimensionsKey] === histogram else { return }
+                dimensions.removeValue(forKey: dimensionsKey)
+                store[histogram.name] = .timeHistogramWithLabels(labelNames, dimensions, buckets)
+            default:
+                return
+            }
+        }
+    }
+
+    public func destroyValueHistogram(_ histogram: ValueHistogram) {
+        self.box.withLockedValue { store in
+            switch store[histogram.name] {
+            case .valueHistogram(let storedHistogram):
+                guard storedHistogram === histogram else { return }
+                store.removeValue(forKey: histogram.name)
+            case .valueHistogramWithLabels(let labelNames, var dimensions, let buckets):
+                let dimensionsKey = LabelsKey(histogram.labels)
+                guard dimensions[dimensionsKey] === histogram else { return }
+                dimensions.removeValue(forKey: dimensionsKey)
+                store[histogram.name] = .valueHistogramWithLabels(labelNames, dimensions, buckets)
+            default:
+                return
+            }
+        }
+    }
+
+    // MARK: Emitting
+
+    public func emit(into buffer: inout [UInt8]) {
+        let metrics = self.box.withLockedValue { $0 }
```

Comment:
I know this is the NIO type, but tbh for such there should be a get() on it, WDYT?